File: BO20426122_capstone/frontend_online_reatil_store/Frontend/online-retail-store/src/app/services/product.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Product } from '../models/Product.model';

@Injectable({
  providedIn: 'root'
})
export class ProductService {
  private baseUrl = 'http://localhost:3333/api/products';

  constructor(private http: HttpClient) {}

  addProduct(product: Product): Observable<Product> {
    return this.http.post<Product>(`${this.baseUrl}/addProduct`, product);
  }

  getProduct(productId: number): Observable<Product> {
    return this.http.get<Product>(`${this.baseUrl}/${productId}`);
  }

  updateProduct(productId: number, product: Product): Observable<Product> {
    return this.http.put<Product>(`${this.baseUrl}/updateProduct/${productId}`, product);
  }

  deleteProduct(productId: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${productId}`);
  }

  getProducts(): Observable<Product[]> {
    return this.http.get<Product[]>(`${this.baseUrl}`);
  }
}

File: BO20426122_capstone/backend_online_retail_store/Online_retail_store/Products/src/main/java/com/onlineretailstore/products/service/ProductService.java
package com.onlineretailstore.products.service;

import java.util.List;

import com.onlineretailstore.products.entity.Product;

public interface ProductService {

	public Product addProduct(Product product);
	public void deleteProduct(Integer productId);
	public Product updateProduct(Integer productId, Product product);
	public Product getProduct(Integer productId);
	public List<Product> getAllProducts();
}
